### PR TITLE
intel-oneapi-mkl: Add shared library option + static lib fix

### PR DESCRIPTION
**Pull request by he-b**
Added option for shared libraries.

Also fixed circular dependency when only using static libraries.

### CODE REVIEW

**Comment by bernhardkaindl** [comment]
@he-b: Run `flake8` on your package.py (and even better `spack style <path/to/package.py>`)
```py
flake8-3.8.2-ag7jbfdh2m25uv6ignnwfkl7e7dioirh
  var/spack/repos/builtin/packages/intel-oneapi-mkl/package.py:68: [E501] line too long (94 > 88 characters)
  var/spack/repos/builtin/packages/intel-oneapi-mkl/package.py:70: [E501] line too long (92 > 88 characters)
  flake8 found errors
```
When you amend your commit using `git commit --amend` (you need to use `git push -f` then), prefix the commit message with
```
intel-oneapi-mkl:
```

File: var/spack/repos/builtin/packages/intel-oneapi-mkl/package.py [hunk 1]
```diff
@@ -61,8 +62,12 @@ def headers(self):
 
     @property
     def libs(self):
+        shared = True if '+shared' in self.spec else False
         mkl_libs = [self.xlp64_lib('libmkl_intel'), 'libmkl_sequential', 'libmkl_core']
         libs = find_libraries(mkl_libs,
-                              join_path(self.component_path, 'lib', 'intel64'))
+                              join_path(self.component_path, 'lib', 'intel64'), shared=shared)
+        if not shared:
+            libs = LibraryList(['-Wl,--start-group'] + libs.libraries + ['-Wl,--end-group'])
```
Comment (bernhardkaindl) [inline var/spack/repos/builtin/packages/intel-oneapi-mkl/package.py]:
```suggestion
                              join_path(self.component_path, 'lib', 'intel64'),
                                        shared=shared)
        if not shared:
            libs = LibraryList(['-Wl,--start-group'] + libs.libraries +
                               ['-Wl,--end-group'])
```

**Review by bernhardkaindl — CHANGES_REQUESTED** [review]
I assume this could make the flake8 tool happy (you can install it with pip3 as well).

You can apply this suggestion witth button next to, but run it yourself after pulling the commit.

**Comment by rscohn2** [comment]
@he-b: I have a pending PR that overlaps with your changes because it adds the libraries for scalapack interfaces. Could you submit these changes as a PR against that branch? Then we could request the changes be merged with both commits. I usually add tests and can add a test for static linking.
https://github.com/spack/spack/pull/26256

**Comment by he-b** [comment]
@rscohn2 thank you for your comments, I created a PR  in your branch as you requested.

**Pull request closed without merging**